### PR TITLE
Upgrade Java version in community-build

### PR DESCRIPTION
**Pull request by nicolasstucki**
[skip test]
[skip test_windows_fast]
[skip test_sbt]
[skip mima]
[skip docs]

### CODE REVIEW

**Comment by nicolasstucki** [comment]
`community_build_a` and `community_build_b` fail on Java 16. I will open an issue with the description of the failures.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -350,13 +358,17 @@ jobs:
...
+           && !contains(github.event.pull_request.body, '[skip community_build_c]')
          )
          || (
            github.event_name == 'workflow_dispatch'
            && github.repository == 'lampepfl/dotty'
          )"
 
     steps:
+      - name: Set JDK 17 as default
+        run: echo "/usr/lib/jvm/java-17-openjdk-amd64/bin" >> $GITHUB_PATH
```
Comment (smarter) [inline .github/workflows/ci.yaml:370]: I don't think Java 17 is installed on the docker image we're currently running, so this might not do anything

Comment (smarter) [reply .github/workflows/ci.yaml:370]: The Dockerfile was updated in October 2021: https://github.com/lampepfl/dotty-ci/pull/19, but the image we're running is from March 2021: https://github.com/lampepfl/dotty/blob/main/.github/workflows/ci.yaml#L37

Comment (nicolasstucki) [reply .github/workflows/ci.yaml]: I changed it to JDK 16. We can update it when we update the image.

How do we update that image?

Comment (smarter) [reply .github/workflows/ci.yaml]: To push a new image, see instructions in https://github.com/lampepfl/dotty-ci#dotty-docker-image, then the ci.yaml file can be updated to point to the new one.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -256,13 +256,17 @@ jobs:
...
            && !contains(github.event.pull_request.body, '[skip community_build]')
+           && !contains(github.event.pull_request.body, '[skip community_build_a]')
          )
          || (
            github.event_name == 'workflow_dispatch'
            && github.repository == 'lampepfl/dotty'
          )"
 
     steps:
+      - name: Set JDK 15 as default
```
Comment (smarter) [inline .github/workflows/ci.yaml:267]: I don't think the image has JDK 15 either, just JDK 16 and JDK 8 as the default, so this probably doesn't do anything.

**Pull request closed without merging**